projects/schematics/src/add-spartacus/spartacus-feature-toggles.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  Rule,
+  SchematicContext,
+  SchematicsException,
+  Tree,
+} from '@angular-devkit/schematics';
+import { SourceFile } from 'ts-morph';
+import {
+  PROVIDE_FEATURE_TOGGLES_FUNCTION,
+  SPARTACUS_CORE,
+  SPARTACUS_FEATURES_NG_MODULE,
+  addModuleProvider,
+  getProjectTsConfigPaths,
+  getSpartacusFeaturesModule,
+  saveAndFormat,
+} from '../shared';
+import { Schema as SpartacusOptions } from './schema';
+
+export function addFeatureToggles(options: SpartacusOptions): Rule {
+  return (tree: Tree, context: SchematicContext): Tree => {
+    if (options.debug) {
+      context.logger.info(`⌛️ Adding Feature Toggles...`);
+    }
+
+    const { buildPaths } = getProjectTsConfigPaths(tree, options.project);
+
+    if (!buildPaths.length) {
+      throw new SchematicsException(
+        `Could not find any tsconfig file. Can't find ${SPARTACUS_FEATURES_NG_MODULE}.`
+      );
+    }
+
+    const basePath = process.cwd();
+    for (const tsconfigPath of buildPaths) {
+      const spartacusFeaturesModule = getSpartacusFeaturesModule(
+        tree,
+        basePath,
+        tsconfigPath
+      );
+      if (spartacusFeaturesModule) {
+        _addFeatureToggles(spartacusFeaturesModule);
+        break;
+      }
+    }
+
+    if (options.debug) {
+      context.logger.info(`✅ Feature Toggles added.`);
+    }
+    return tree;
+  };
+}
+/**
+ * Creates and adds a spartacus feature toggles with all of them enabled.
+ */
+function _addFeatureToggles(sourceFile: SourceFile): void {
+  const featureTogglesProvider = createFeatureTogglesProvider();
+  addModuleProvider(sourceFile, {
+    import: [
+      {
+        moduleSpecifier: SPARTACUS_CORE,
+        namedImports: [PROVIDE_FEATURE_TOGGLES_FUNCTION],
+      },
+    ],
+    content: featureTogglesProvider,
+  });
+
+  saveAndFormat(sourceFile);
+}
+
+/**
+ * Returns a provider for all existing feature toggles with value `true`.
+ */
+function createFeatureTogglesProvider(): string {
+  const {
+    defaultFeatureToggles,
+  } = require('../feature-toggles.copied-from-core-lib');
```

Comment:
`nitpick:` I have mixed feelings about this path because, from a code reader's perspective, it's a bit magical and doesn't refer to an existing file "at this point", but I guess we can't create one const `DESTINATION` to use here and in `copy-feature-toggles.js`.